openmc/filter.py

```diff
@@ -77,6 +77,23 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self == other
 
+    def __gt__(self, other):
+        if self.type != other.type:
+            if self.type in _FILTER_TYPES and other.type in _FILTER_TYPES:
+                delta = _FILTER_TYPES.index(self.type) - \
+                        _FILTER_TYPES.index(other.type)
+                return True if delta > 0 else False
+            else:
+                return False
+        else:
+            if 'energy' in self.type and 'energy' in other.type:
```

Comment:
Sure - the short of it is this is needed when merging filters with energy filters. In particular, the order of the tally bin array concatenation is dependent on the result of this comparison.